pydm/widgets/timeplot.py

```diff
@@ -231,24 +366,133 @@ def setCurves(self, new_list):
+
+            self.timeplot_display.show_mouse_coordinates(mouse_point.x(), mouse_point.y())
+            self.vertical_crosshair_line.setPos(mouse_point.x())
+            self.horizontal_crosshair_line.setPos(mouse_point.y())
+
+    def enableCrosshair(self, is_enabled, vertical_angle=90, horizontal_angle=0, vertical_movable=False,
+                        horizontal_movable=False):
+        self.timeplot_display.show_mouse_coordinates(DEFAULT_X_MIN, DEFAULT_Y_MIN)
+        if is_enabled:
+            self.vertical_crosshair_line = InfiniteLine(pos=DEFAULT_X_MIN, angle=vertical_angle,
+                                                        movable=vertical_movable)
+            self.horizontal_crosshair_line = InfiniteLine(pos=DEFAULT_Y_MIN, angle=horizontal_angle,
+                                                          movable=horizontal_movable)
+
+            self.plotItem.addItem(self.vertical_crosshair_line)
+            self.plotItem.addItem(self.horizontal_crosshair_line)
+            self.crosshair_movement_proxy = SignalProxy(self.plotItem.scene().sigMouseMoved, rateLimit=60,
+                                                        slot=self.mouseMoved)
+        else:
+            if self.vertical_crosshair_line:
+                self.plotItem.removeItem(self.vertical_crosshair_line)
+            if self.horizontal_crosshair_line:
+                self.plotItem.removeItem(self.horizontal_crosshair_line)
+            if self.crosshair_movement_proxy:
+                self.crosshair_movement_proxy.disconnect()
+
     def getBufferSize(self):
         return int(self._bufferSize)
 
     def setBufferSize(self, value):
         if self._bufferSize != int(value):
-            self._bufferSize = max(int(value), 1)
+            # Originally, the bufferSize is the max between the user's input and 1, and 1 doesn't make sense.
+            # So, I'm comparing the user's input with the minimum buffer size, and pick the max between the two
+            self._bufferSize = max(int(value), MINIMUM_BUFFER_SIZE)
             for curve in self._curves:
                 curve.setBufferSize(value)
 
     def resetBufferSize(self):
-        if self._bufferSize != 1200:
-            self._bufferSize = 1200
+        if self._bufferSize != MINIMUM_BUFFER_SIZE:
+            self._bufferSize = MINIMUM_BUFFER_SIZE
             for curve in self._curves:
                 curve.resetBufferSize()
 
     bufferSize = Property("int", getBufferSize,
                           setBufferSize, resetBufferSize)
 
+    def setContinuationTimer(self, timeout_milliseconds):
+        if timeout_milliseconds >= 0:
+            self.continuation_timeout = timeout_milliseconds
+            self.continuation_timer.timeout.connect(self.handle_continuation_timer_timeout)
+            self.continuation_timer.start(self.continuation_timeout)
+        else:
+            self.continuation_timer.stop()
+        self.redraw_timer.start()
+
+    def pausePlotting(self):
```

Comment:
This seems like it should be a universal `PyDM` plotting option. Instead of pausing a timer, could we just put a bypass in the `redrawPlot` function on `BasePlot`?

Reply:
I tried to use _needs_redraws, but it turned out this variable would be set back to True whenever the curve's data has changed (due to the set_needs_redraw() slot being connected to the data_changed signal of the same curve). So, I'm keeping the timer start/stop for pausing.